contracts/NFTMarket.sol

```diff
@@ -488,12 +489,46 @@ contract NFTMarket is
         listedTokenIDs[address(_tokenAddress)].remove(_id);
         _updateListedTokenTypes(_tokenAddress);
 
-        skillToken.transferFrom(msg.sender, taxRecipient, taxAmount);
-        skillToken.transferFrom(
-            msg.sender,
-            listing.seller,
-            finalPrice.sub(taxAmount)
-        );
+
+        (uint256 fromInGameOnlyFunds, uint256 fromTokenRewards, uint256 fromUserWallet) =
+            ICryptoBlades(taxRecipient).getSkillToSubtract(
+                ICryptoBlades(taxRecipient).inGameOnlyFunds(msg.sender),
+                ICryptoBlades(taxRecipient).getTokenRewardsFor(msg.sender),
+                finalPrice
+            );
+
+        if (fromInGameOnlyFunds >= finalPrice) {
+            // If Buyer has enough inGameOnlyFunds
+            ICryptoBlades(taxRecipient).transferInGameOnlyFunds(msg.sender, taxRecipient, taxAmount);
+            ICryptoBlades(taxRecipient).transferInGameOnlyFunds(msg.sender, listing.seller, finalPrice.sub(taxAmount));
+
+        } else if (fromInGameOnlyFunds.add(fromTokenRewards) >= finalPrice) {
+            // If Buyer has funds in both inGameOnlyFunds and tokenRewards
+            uint256 taxAmountInGameOnlyFunds = fromInGameOnlyFunds.mul(taxAmount).div(finalPrice);
+            ICryptoBlades(taxRecipient).transferInGameOnlyFunds(msg.sender, taxRecipient, taxAmountInGameOnlyFunds);
+            ICryptoBlades(taxRecipient).transferInGameOnlyFunds(msg.sender, listing.seller, fromInGameOnlyFunds.sub(taxAmountInGameOnlyFunds));
+
+            uint256 taxAmountTokenRewards = taxAmount.sub(taxAmountInGameOnlyFunds);
+            ICryptoBlades(taxRecipient).transferTokenRewards(msg.sender, taxRecipient, taxAmountTokenRewards);
+            ICryptoBlades(taxRecipient).transferTokenRewards(msg.sender, listing.seller, fromTokenRewards.sub(taxAmountTokenRewards));
+        } else {
+            uint256 taxAmountInGameOnlyFunds = fromInGameOnlyFunds.mul(taxAmount).div(finalPrice);
+            ICryptoBlades(taxRecipient).transferInGameOnlyFunds(msg.sender, taxRecipient, taxAmountInGameOnlyFunds);
+            ICryptoBlades(taxRecipient).transferInGameOnlyFunds(msg.sender, listing.seller, fromInGameOnlyFunds.sub(taxAmountInGameOnlyFunds));
+
+            uint256 taxAmountTokenRewards = fromTokenRewards.mul(taxAmount).div(finalPrice);
+            ICryptoBlades(taxRecipient).transferTokenRewards(msg.sender, taxRecipient, taxAmountTokenRewards);
+            ICryptoBlades(taxRecipient).transferTokenRewards(msg.sender, listing.seller, fromTokenRewards.sub(taxAmountTokenRewards));
+
+            uint256 taxAmountUserWallet = taxAmount.sub(taxAmountInGameOnlyFunds).sub(taxAmountTokenRewards);
```

Comment:
Cast `taxRecipient` just once and store the result in a local variable.

Also, we do not want to give token rewards or IGO to `taxRecipient` since that's gonna be the main Cryptoblades contract.